x-pack/solutions/observability/plugins/metrics_data_access/common/inventory_models/shared/metrics/snapshot/network_traffic.ts

```diff
@@ -88,18 +88,63 @@ export const networkTrafficWithInterfacesWithFilter = (
       },
     },
   },
-  [`${id}_deriv`]: {
-    derivative: {
-      buckets_path: `${id}_dimension>${id}_sum_of_interfaces`,
+  [id]: {
+    bucket_script: {
+      buckets_path: { value: `${id}_deriv[normalized_value]` },
+      script: {
+        source: 'params.value > 0.0 ? params.value : 0.0',
+        lang: 'painless',
+      },
       gap_policy: 'skip',
-      unit: '1s',
     },
   },
+});
+
+export const networkTrafficSnapshotWithFilter = (
+  id: string,
+  metricField: string,
+  interfaceField: string,
+  filter: estypes.QueryDslQueryContainer
+): MetricsUIAggregation => ({
+  [`${id}_dimension`]: {
+    filter,
+    aggs: {
+      [`${id}_interfaces`]: {
+        terms: { field: interfaceField },
+        aggregations: {
+          [`${id}_interface_max`]: { max: { field: metricField } },
+          [`${id}_interface_min`]: { min: { field: metricField } },
+          [`${id}_interface_diff`]: {
+            bucket_script: {
+              buckets_path: { max: `${id}_interface_max`, min: `${id}_interface_min` },
+              script: 'params.max - params.min',
```

Comment:
I'm not sure this will calculate the rate correctly for counters. It ignores the time distribution, which is fine for gauges, but not for counters. The derivative aggregation, on the other hand, calculates the difference between consecutive buckets, which is why it gives an accurate rate over time.